client/my-sites/site-settings/publishing-tools/hooks/use-switch-post-by-voice-mutation.tsx

```diff
@@ -0,0 +1,63 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { getCachePostByVoiceKey } from 'calypso/my-sites/site-settings/publishing-tools/hooks/get-cache-post-by-voice-key';
+import { getPostByVoicePath } from 'calypso/my-sites/site-settings/publishing-tools/hooks/get-post-by-voice-path';
+import { parsePostByVoiceResponse } from 'calypso/my-sites/site-settings/publishing-tools/hooks/parse-post-by-voice-response';
+import type { PostByVoice } from 'calypso/my-sites/site-settings/publishing-tools/types/post-by-voice';
+import type { PostByVoiceResponse } from 'calypso/my-sites/site-settings/publishing-tools/types/post-by-voice-response';
+
+export const useSwitchPostByVoiceMutation = ( siteId: number | null ) => {
+	const queryClient = useQueryClient();
+	const queryKey = getCachePostByVoiceKey( siteId );
+
+	return useMutation( {
+		mutationFn: async ( value: boolean ) => {
+			let response: PostByVoiceResponse;
+			if ( value ) {
+				response = await await wpcom.req.post( {
+					method: 'POST',
+					path: getPostByVoicePath( siteId ),
+					apiNamespace: 'wpcom/v2',
+				} );
+			} else {
+				response = await await wpcom.req.del( {
+					method: 'DELETE',
+					path: getPostByVoicePath( siteId ),
+					apiNamespace: 'wpcom/v2',
+				} );
+			}
```

Comment:
```suggestion
		mutationFn: async ( shouldEnable: boolean ) => {
			const response: PostByVoiceResponse = await wpcom.req.post( {
				method: shouldEnable ? 'POST' : 'DELETE',
				path: getPostByVoicePath( siteId ),
				apiNamespace: 'wpcom/v2',
			} );
```
The `wpcom.req` API is quite clunky, and it might look counterintuitive that we can call `wpcom.req.post` and still override the method by passing a `method` param, _but we can_. Since there are no other differences than the request method between the enable/disable requests, I think something like my suggestion would be preferable.